src/entry-client.js

```diff
@@ -32,35 +33,84 @@ export default function initializeClient(createApp, clientOpts) {
         // dynamic modules until Vuex implements a hasModule() type method.  See:
         //   https://github.com/vuejs/vuex/issues/833
         //   https://github.com/vuejs/vuex/pull/834
-        const registeredModules = {};
+        const registeredModules = [];
+
+        // Allow a function to be passed that can generate a route-aware
+        // module name
+        const getModuleName = (c, route) => (
+            typeof c.vuex.moduleName === 'function' ?
+                c.vuex.moduleName(route) :
```

Comment:
Components can pass in a function that will be passed the route and should return a route-aware module name